### PR TITLE
First draft updating Trusted Cluster

### PR DESCRIPTION
**Pull request by benarent**
I've updated the images for trusted clusters, plus I've added a MSP example that shows many more clusters. 

Once this PR Is reviewed I'll back port it the other versions / 4.1 .

This starts to expand info for https://github.com/gravitational/teleport/issues/3007

### CODE REVIEW

File: docs/4.0/trustedclusters.md [hunk 1]
```diff
@@ -55,17 +67,20 @@ $ tsh ssh --cluster=east host
 $ tsh clusters
 ```
 
-Trusted clusters also have their own restrictions on user access, i.e.
+Leaf clusters also have their own restrictions on user access, i.e.
 _permissions mapping_ takes place.
 
+**Once Connected has been established, it's easy to switch from the "main" root cluster**
```
Comment (webvictim) [inline docs/4.0/trustedclusters.md]: Either "Once connected" or "Once connection has been established"

File: docs/4.0/trustedclusters.md [hunk 2]
```diff
@@ -219,6 +236,17 @@ role name and use reference it to name the local role:
 **NOTE:** The regexp matching is activated only when the expression starts
 with `^` and ends with `$`
 
+
+### Trusted Cluster UI
+For customers using Teleport Pro, they can easily configure _leaf_ nodes using the
```
Comment (webvictim) [inline docs/4.0/trustedclusters.md]: I don't think we sell Teleport Pro any more so this should reference Enterprise. I also don't think that the trusted cluster UI is an Enterprise-only feature - probably need to double check.

File: docs/4.0/trustedclusters.md [hunk 3]
```diff
@@ -219,6 +236,17 @@ role name and use reference it to name the local role:
...
 
+
+### Trusted Cluster UI
+For customers using Teleport Pro, they can easily configure _leaf_ nodes using the 
+Teleport Proxy UI. 
+
+**Creating Trust from the Leaf node to the root node.**
+![Tunnels](img/trusted-clusters/setting-up-trust.png)
+
+**The root Cluster showing the Cluster switching UI.**
```
Comment (webvictim) [inline docs/4.0/trustedclusters.md]: Inconsistency with other use of `Cluster` vs `cluster` in the docs here. I'd probably make it all lower case `cluster` for ease.

File: docs/4.0/trustedclusters.md [hunk 4]
```diff
@@ -240,7 +268,7 @@ Now an admin from the main cluster can now see and access the "east" cluster:
 
 ```bsh
 # login into the main cluster:
-$ tsh --proxy=proxy.main login admin
+$ tsh --proxy=main.example.com login admin
```
Comment (webvictim) [inline docs/4.0/trustedclusters.md]: If we're using `main.example.com` here and elsewhere in the docs and `<main-proxy-address>` above it's probably a little confusing; I'd stick with `main.example.com` throughout.

File: docs/4.0/trustedclusters.md [hunk 5]
```diff
@@ -192,8 +209,8 @@ spec:
     - remote: "admin"
       local: [local-admin]
   token: "join-token"
-  tunnel_addr: main.example.com:3024
-  web_proxy_addr: main.example.com:3080
+  tunnel_addr: <main-proxy-address>:3024
```
Comment (webvictim) [inline docs/4.0/trustedclusters.md]: I'd probably put this back to `main.example.com` to be consistent with the rest of the docs.

**Review by webvictim — APPROVED** [review]
I like this in general. I'm a little wary of the fact that the Teleport code itself hasn't been updated to replace "trusted cluster" with "root cluster" and "trusting cluster"/"remote cluster" with "leaf cluster" yet. I don't know whether we're actually planning to do that, as we'd likely have to change all the references:

- (trusted cluster) `tctl rm tc/main` -> (root cluster) `tctl rm rc/main`
- (remote cluster) `tctl rm rc/east` -> (leaf cluster) `tctl rm lc/east`

This would get confusing because `remote cluster/rc` would become `root cluster/rc` and have a totally different meaning. Not sure if I'm overthinking this though.

@klizhentas Any thoughts on whether we're going to change the references Teleport uses internally from trusted/(trusting|remote) to root/leaf or not?

File: docs/4.0/trustedclusters.md (hunk 2 above)
Comment (benarent) [reply docs/4.0/trustedclusters.md]: Good call, yeah, I checked it's a Ent only feature in 4.0

**Comment by benarent** [comment]
Good catch @webvictim I didn't realise we had it documented here https://gravitational.com/teleport/docs/admin-guide/#disabling-trust

**Comment by benarent** [comment]
I'm going to merge this down so it's included in the 4.1 release. I'll make another issue to discuss the admin guide `tctl` options.

**Pull request merged**